### ISSUE TITLE
Subsetting GeoArray

### ISSUE THREAD

**Issue by KrOstir** [evetion/GeoArrays.jl#73]
When a GeoArray is subset by index, only the array is returned. However, a subset of a GeoArray - either spatial or band - is a GeoArray.

Here is an example.

```
fn = download("https://bitbucket.org/earthobservation/julia-remote-sensing/raw/ccea113bf43ba71d4d1c6f02e060fd2923c982b2/data/KR_S2B_MSIL2A_20210614_20.tif")
geoarray = GeoArrays.read(fn)

ga_band = geoarray[:,:,7] # Matrix{UInt16}
```

The subset is an array.

I have written a quick and dirty - I am a Julia beginner - function to make the subset as a GeoArray. 

```
function ga_subset(ga, bands, ind)
	afm = ga.f
	r_min = ind[1]
	r_max = ind[3]
	c_min = ind[2]
	c_max = ind[4]
	sub_afm_l = afm.linear
	sub_a = ga.A[r_min:r_max,c_min:c_max,bands]
	sub_afm_t = GeoArrays.coords(ga, [r_min,c_min])
	sub_f = CoordinateTransformations.AffineMap(sub_afm_l, sub_afm_t)
	sub_crs = ga.crs
	sub_ga = GeoArray(sub_a, sub_f, sub_crs)
    return sub_ga
end
```

This works for row and column and band subsets. It could be rewritten also to use geographical coordinates (compute index from coords).

```
ga_sub = ga_subset(geoarray, [1,7], [200,200,400,300]);
typeof(ga_sub) # 201x101x2 Array{UInt16, 3} with AffineMap([20.0 0.0; 0.0 -20.0], [448980.0, 5.12002e6]) ...
```

GeoArray slicing is of course complex, but for simple subsetting adding methods for band slicing and spatial slicing could be beneficial.

**Comment by evetion** [comment]
I'd like this as well. Subsetting the `AffineMap` would take some thorough thinking and testing though.

**Comment by evetion** [comment]
Fixed by #74
